src/commands/prompt/run.ts

```diff
@@ -97,34 +50,57 @@ async function createSession(options: PromptOptions): Promise<SessionContext> {
     }
   }
 
+  const messages: Message[] = [];
+
+  if (initialPrompt) {
+    messages.push({
+      role: 'user',
+      content: initialPrompt,
+    });
+  }
+
   const config = {
     model: options.model ?? initialConfig.model,
     apiKey: initialConfig.apiKey,
     systemPrompt: initialConfig.systemPrompt,
     ...style,
   };
 
-  output.outputVerbose(`Using model: ${config.model}`);
+  if (options.file) {
+    const { fileContextPrompt, costWarning, costInfo } = handleInputFile(
+      options.file,
+      config,
+      provider,
+    );
+    messages.push(fileContextPrompt);
+    if (costWarning) {
+      sessionFeedback.fileCostWarning = costWarning;
+    }
+    if (costInfo) {
+      sessionFeedback.fileCostInfo = costInfo;
+    }
+  }
 
   return {
     config,
     provider,
-    messages: [],
-    totalUsage: { inputTokens: 0, outputTokens: 0, requests: 0 },
+    messages,
+    sessionFeedback,
   };
 }
 
-async function handleMessage(session: SessionContext, message: string) {
-  output.outputAiProgress('Thinking...');
-
-  session.messages.push({ role: 'user', content: message });
-  const response = await session.provider.getChatCompletion(session.config, session.messages);
-  session.totalUsage = combineUsage(session.totalUsage, response.usage);
-
-  output.clearLine();
-  output.outputVerbose(`Response Object: ${JSON.stringify(response.response, null, 2)}`);
-
-  const outputParams = getOutputParams(session, response);
-  output.outputAi(response.messageText ?? '(null)', outputParams);
-  session.messages.push({ role: 'assistant', content: response.messageText ?? '' });
+export async function getChatCompletion(
```

Comment:
What is the purpose of this functions. Seems like a very thin wrapper. Do we need it?

Reply:
removed